### DISCUSSION TITLE
Request matching headers

### DISCUSSION THREAD

**Discussion by hurena-BR** [fauxauldrich/camouflage#181] [Q&A] [answered]
Is there any way to do a request matching making sure a specific header is specified? I'm trying to send a different response if the request does not include a specific header. 

I already tried using:  {{#if request.headers.header-name}}

**Comment by shubhendumadhukar** [comment, marked as answer]
There is a bug in Camouflage that prevents request matching using headers. I have made the required changes and it should work in `0.8.2` in following manner.
```
{{#if (capture from='headers' key='Authorization') }}
HTTP/1.1 200 OK
Content-Type: application/json

{
    "response": "response if auth header is present."
}
{{else}}
HTTP/1.1 401 Unauthorized
Content-Type: application/json

{
    "response": "response if no auth header present."
}
{{/if}}
```
If you want to validate a given header against a specific value, the mock file would be as shown below:
```
{{#is (capture from='headers' key='Authorization') 'Basic c2h1YmhlbmR1Om1hZGh1a2Fy' }}
HTTP/1.1 200 OK
Content-Type: application/json

{
    "response": "response if auth header is present."
}
{{else}}
HTTP/1.1 401 Unauthorized
Content-Type: application/json

{
    "response": "response if no auth header present."
}
{{/is}}
```
Note that above snippets will work with `0.8.2`. If you need it to work now, you can use `code` helper as shown below (You will need to enable injection in config.yml)
**Code**
```
HTTP/1.1 200 OK
Content-Type: application/json

{{#code}}
(()=>{
    const authHeader = request.get('Authorization');
    if(authHeader){
        return {
            status: 200,
            headers: {
                'Content-Type': 'application/json'
            },
            body: `{
                "response": "response if auth header is present."
            }`
        };
    }else{
        return {
            status: 401,
            headers: {
                'Content-Type': 'application/json'
            },
            body: `{
                "response": "response if no auth header present."
            }`
        };
    }
})();
{{/code}}
```

**Reply by hurena-BR** [reply]
Awesome! Thanks! Using the code helper, I was able to make it work, I had already tried using the code helper but was getting an error related to Json formatting, all I needed to do was enabling injection in config.yaml.

**Reply by shubhendumadhukar** [reply]
I have released 0.8.2. Both the approaches should work now.
